frontend/src/metabase/hoc/Favicon.jsx

```diff
@@ -1,43 +1,14 @@
-import { useEffect } from "react";
+import { useFavicon as useFaviconMantineHook } from "@mantine/hooks";
 
-import MetabaseSettings from "../lib/settings";
+import { useSelector } from "metabase/lib/redux";
+import { getSetting } from "metabase/selectors/settings";
 
-const DEFAULT_FAVICON = () => MetabaseSettings.get("application-favicon-url");
 export const LOAD_COMPLETE_FAVICON = "app/assets/img/blue_check.png";
 
-const resolveFavicon = (setterOrGetter, props) => {
-  if (typeof setterOrGetter === "string") {
-    return setterOrGetter;
-  } else if (typeof setterOrGetter === "function") {
-    const result = setterOrGetter(props);
-    if (result == null) {
-      return DEFAULT_FAVICON();
-    } else if (result instanceof String || typeof result === "string") {
-      return result;
-    }
-  }
-};
-
-/**
- * @deprecated HOCs are deprecated
- */
-const withFavicon = faviconSetterOrGetter => ComposedComponent => {
-  const WithFavicon = props => {
-    const favicon = resolveFavicon(faviconSetterOrGetter, props);
+export const useFavicon = ({ favicon }) => {
+  const defaultFavicon = useSelector(state =>
+    getSetting(state, "application-favicon-url"),
+  );
 
-    useEffect(() => {
-      document.querySelector('link[rel="icon"]')?.setAttribute("href", favicon);
-      return () => {
```

Comment:
This is the code that set back the default favicon, I think it's easier to extract this into our custom hook rather than using the mantine hook, as it doesn't seem like that supports resetting the original favicon

Reply:
gotcha. let me see if i can do something about this